openvdb_points/tools/PointCount.h

```diff
@@ -261,6 +282,49 @@ Index64 inactiveGroupPointCount(const PointDataTreeT& tree, const Name& name, co
 }
 
 
+template <typename PointDataTreeT>
+Index64 getPointOffsets(std::vector<Index64>& pointOffsets, const PointDataTreeT& tree,
+                     const std::vector<Name>& includeGroups, const std::vector<Name>& excludeGroups,
+                     const bool inCoreOnly)
+{
+    typedef typename PointDataTreeT::LeafNodeType LeafNode;
+    typedef typename LeafNode::IndexOnIter IndexOnIter;
+
+    const bool useGroup = includeGroups.size() > 0 || excludeGroups.size() > 0;
+
+    tree::LeafManager<const PointDataTreeT> leafManager(tree);
+    const size_t leafCount = leafManager.leafCount();
+
+    pointOffsets.reserve(leafCount);
+
+    Index64 pointOffset = 0;
+    for (size_t n = 0; n < leafCount; n++)
+    {
+        const LeafNode& leaf = leafManager.leaf(n);
+
+        pointOffsets.push_back(pointOffset);
+#ifndef OPENVDB_2_ABI_COMPATIBLE
+        // skip out-of-core leafs
+        if (inCoreOnly && leaf.buffer().isOutOfCore())  continue;
+#else
+        (void) inCoreOnly; // unused variable
+#endif
+
+        if (useGroup) {
+            IndexOnIter iter = leaf.beginIndexOn();
+            MultiGroupFilter::Data data(includeGroups, excludeGroups);
+            const MultiGroupFilter filter = MultiGroupFilter::create(leaf, data);
+            FilterIndexIter<IndexOnIter, MultiGroupFilter> filterIndexIter(iter, filter);
+            pointOffset += iterCount(filterIndexIter);
+        }
+        else {
+            pointOffset += leaf.onPointCount();
+        }
+    }
+    return pointOffset;
+}
+
+
```

Comment:
The first implementation of this basic function allowed for accumulated leaf point counts as well as point offsets (essentially starting the first element with leaf.pointCount() and accumulating). Is that something you'd consider adding here? I've found it useful in some cases.